File: examples/python/core/parallel/check-failures.py
from aws_durable_execution_sdk_python import BatchResult

result: BatchResult = context.parallel(functions)

if result.failure_count > 0:
    # Some branches failed
    return {
        "status": "partial_failure",
        "successful": result.get_results(),
        "failed_count": result.failure_count,
    }

# All branches succeeded
return {
    "status": "success",
    "results": result.get_results(),
}

File: examples/java/core/parallel/check-failures.java
// Use map to run branches and check for failures
MapResult<String> result = ctx.map("parallel-work", items, String.class,
    (item, index, childCtx) -> childCtx.step("step-" + index, String.class, stepCtx -> process(item)));

if (!result.allSucceeded()) {
    return Map.of(
        "status", "partial_failure",
        "successful", result.succeeded(),
        "failed_count", result.failed().size()
    );
}

return Map.of(
    "status", "success",
    "results", result.succeeded()
);
